### PR TITLE
bluetooth: services: bas: Move log module to bas and guard BT_BAS_BLS

### PR DESCRIPTION
**Pull request by bjarki-andreasen**
The bas service can be used without BT_HCI=y. However, if BT_HCI is not set, the BT Kconfig.logging is not included, which results in

module = BT_BAS
module-str = BAS
source "subsys/logging/Kconfig.template.log_config"

not being defined. Furthermore, BT_BAS_BLS depends on BT_BAS, so move the log module definition to Kconfig.bas, guarded by BT_BAS, and guard BT_BAS_BLS behind BT_BAS.

This issue was detected by a project setting BT_HCI_IPC=y, LOG=y and BT=y

Lastly, remove redundant LOG_LEVEL define.

### CODE REVIEW

**Review by Thalley — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by kapi-no — CHANGES_REQUESTED** [review]
The change is not needed. 

You cannot select the `BT_BAS` Kconfig without the having the `BT_CONN` Kconfig enabled:

```
menu "GATT Services"
	depends on BT_CONN

rsource "Kconfig.dis"

rsource "Kconfig.cts"

rsource "Kconfig.hrs"

rsource "Kconfig.tps"

rsource "nus/Kconfig.nus"

rsource "ias/Kconfig.ias"

rsource "ots/Kconfig"

rsource "bas/Kconfig.bas"

endmenu
```

The `BT_CONN` is a promptless Kconfig and is selected only by the following Kconfigs:

```
 config BT_PERIPHERAL
	bool "Peripheral Role support"
	select BT_BROADCASTER
	select BT_CONN
	default y if BT_HCI_RAW
	help
	  Select this for LE Peripheral role support.

config BT_CENTRAL
	bool "Central Role support"
	select BT_OBSERVER
	select BT_CONN
	default y if BT_HCI_RAW
	help
	  Select this for LE Central role support.
```

These Kconfigs are only available when the `BT_HCI` is enabled.

TLDR: It seems you cannot select `BT_BAS` when `BT_HCI` is disabled without modifying the Bluetooth Kconfigs.

**Comment by bjarki-andreasen** [comment]
Seems this was a false positive due to downstream changes :) Nothing to see here :D

**Comment by Thalley** [comment]
> Seems this was a false positive due to downstream changes :) Nothing to see here :D

`BT_BAS_BLS` should still depend on `BT_BAS` :) So some of the changes here are still valid. Will you continue with those @bjarki-andreasen ?
